plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePageSink.java

```diff
@@ -122,6 +124,13 @@ public void testAllFormats()
                         continue;
                     }
                     config.setHiveCompressionCodec(codec);
+
+                    if (!isSupportedCodec(format, codec)) {
+                        assertThatThrownBy(() -> writeTestFile(config, metastore, makeFileName(tempDir, config)))
+                                .hasMessage("Compression codec " + codec + " not supported for " + format);
```

Comment:
literal `.hasMessage("Compression codec LZ4 not supported for AVRO");` if you choose to inline `isSupportedCodec`

Reply:
Let's keep it as separate method for future extensibility :)